src/webgpu/api/validation/encoding/queries/begin_end.spec.ts

```diff
@@ -9,15 +9,50 @@ TODO:
     - begin 2, end 2
     - }
     - x= {
-        - render pass + occlusion
         - render pass + pipeline statistics
         - compute pass + pipeline statistics
         - }
-- nesting: test whether it's allowed to nest various types of queries
-  (including writeTimestamp inside begin/endable queries).
 `;
 
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
+
+g.test('occlusion_query,begin_end')
+  .desc(
+    `
+Tests that begin/end occlusion query on render pass:
+- begin 0
+- begin 0, end (control case)
+- begin 0, begin 1, end
+- begin 0, end, end
```

Comment:
Nested occlusion query (beginOcclusionQuery, beginOcclusionQuery, endOcclusionQuery, endOcclusionQuery) should be tested either here or in the 'nesting' test below. A separate test is probably ideal, since it tests a different rule, but you can share some code since it will be very simple to describe. In the notation of this test, you could test both:

- begin index 0, begin index 0, end, end
- begin index 0, begin index 1, end, end